### PR TITLE
move torchvision to catalyst[cv]

### PR DESCRIPTION
**Pull request by bagxi**
## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have checked the code style using `catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] I have written tests for all new methods and classes that I created.
- [x] I have written the docstring in Google format for all the methods and classes that I used.
- [ ] I have checked the docs using `make check-docs`.
- [x] I have read I need to click 'Login as guest' to see Teamcity build logs.

### CODE REVIEW

File: catalyst/utils/tools/settings.py [hunk 1]
```diff
@@ -1,13 +1,277 @@
...
-# callbacks
+
+class Settings(FrozenClass):
+    def __init__(
+        self,
+        contrib_required: bool = False,
+        cv_required: bool = False,
+        ml_required: bool = False,
+        nlp_required: bool = False,
+        state_main_metric: str = "loss",
```
Comment (Scitator) [inline catalyst/utils/tools/settings.py]: remove

File: catalyst/utils/tools/settings.py [hunk 2]
```diff
@@ -1,13 +1,277 @@
...
+        stage_train_prefix: str = "train",
+        stage_valid_prefix: str = "valid",
+        stage_infer_prefix: str = "infer",
+        loader_train_prefix: str = "train",
+        loader_valid_prefix: str = "valid",
+        loader_infer_prefix: str = "infer",
```
Comment (Scitator) [inline catalyst/utils/tools/settings.py]: constant

File: catalyst/utils/tools/settings.py [hunk 3]
```diff
@@ -1,13 +1,277 @@
...
+        check_run_num_batch_steps: int = 2,
+        check_run_num_epoch_steps: int = 2,
```
Comment (Scitator) [inline catalyst/utils/tools/settings.py]: remove, constant

File: catalyst/utils/tools/settings.py [hunk 4]
```diff
@@ -1,13 +1,277 @@
...
+        alchemy_logger_required: Optional[bool] = None,
+        neptune_logger_required: Optional[bool] = None,
+        wandb_logger_required: Optional[bool] = None,
+        telegram_logger_token: str = "",
+        telegram_logger_chat_id: str = "",
+        use_lz4: bool = False,
+        use_pyarrow: bool = False,
```
Comment (Scitator) [inline catalyst/utils/tools/settings.py]: write difference between `use_` and `_required`

File: catalyst/utils/tools/settings.py [hunk 5]
```diff
@@ -1,13 +1,277 @@
...
+        albumentations_required: Optional[bool] = None,
+        segmentation_models_required: Optional[bool] = None,
+        use_libjpeg_turbo: bool = False,
+        nmslib_required: Optional[bool] = None,
+        transformers_required: Optional[bool] = None,
```
Comment (Scitator) [inline catalyst/utils/tools/settings.py]: comment group options :)

File: catalyst/contrib/__main__.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 from collections import OrderedDict
 import logging
 
-from catalyst.contrib.scripts import find_thresholds
+from catalyst.contrib.scripts import collect_env, find_thresholds
```
Comment (Scitator) [inline catalyst/contrib/__main__.py:5]: Have you check, that `catalyst-contrib --collect-env` works correctly? should we add this test to `catalyst/bin/tests`?

File: catalyst/contrib/utils/__init__.py [hunk 7]
```diff
@@ -14,17 +16,27 @@
...
+        imread,
+        imwrite,
+        imsave,
+        mask_to_overlay_image,
+        mimread,
+        mimwrite_with_meta,
+        tensor_from_rgb_image,
+        tensor_to_ndimage,
+    )
+except ImportError as ex:
```
Comment (Scitator) [inline catalyst/contrib/utils/__init__.py]: btw, what do you think about changing it to `(ImportError, NameError, AttributeError)`?

File: catalyst/data/scripts/image2embedding.py [hunk 8]
```diff
@@ -9,8 +9,8 @@
 import torch
 from torchvision import transforms
 
+from catalyst.contrib.data.cv import ImageReader
 from catalyst.contrib.models.cv import ResnetEncoder
```
Comment (Scitator) [inline catalyst/data/scripts/image2embedding.py:13]: based on such dependency on `contrib`, should we move such script to `catalyst-contrib`?

File: catalyst/utils/distributed.py [hunk 9]
```diff
@@ -13,6 +13,7 @@
 from torch import nn
 import torch.distributed
 
+from catalyst import __version__
```
Comment (Scitator) [inline catalyst/utils/distributed.py:16]: #TODO: check availability,
```
"check_ddp_wrapped",
    "check_apex_available",
    "check_torch_distributed_initialized",
    "check_slurm_available",
"is_wrapped_with_ddp",
    "is_torch_distributed_initialized",
    "is_slurm_available",
    "is_apex_available",
```
:)

File: requirements/requirements.txt [hunk 10]
```diff
@@ -15,15 +15,9 @@ ipython
 tensorboard>=1.14.0
 tensorboardX
 crc32c>=1.7
+opencv-python-headless
```
Comment (Scitator) [inline requirements/requirements.txt]: why do we need `opencv` still?

Comment (bagxi) [reply requirements/requirements.txt]: tensorboard - https://github.com/catalyst-team/catalyst/blob/9f97ad91c82322a18ac6a5764aaded7fc54720a4/catalyst/contrib/utils/tools/tensorboard.py#L160

File: catalyst/data/scripts/image2embedding.py (hunk 8 above)
Comment (bagxi) [reply catalyst/data/scripts/image2embedding.py:13]: I think no, and those scripts simply wouldn't be loaded if some problems appear during catalyst initialization

btw, we have a similar situation with nlp scripts

File: requirements/requirements.txt (hunk 10 above)
Comment (Scitator) [reply requirements/requirements.txt]: Looks like `SummaryReader` does not use this feature,
- it is used only for scalar logs reading
- by default it does not support image handling
- we don't have a need to read tensorboard image-based logs (usually all you need are scalar one)

What do you think about removing this feature?

File: catalyst/data/scripts/image2embedding.py (hunk 8 above)
Comment (Scitator) [reply catalyst/data/scripts/image2embedding.py:13]: deal, let's think about this improvement later :)

File: requirements/requirements.txt (hunk 10 above)
Comment (bagxi) [reply requirements/requirements.txt]: and what about confusion matrix?

Comment (bagxi) [reply requirements/requirements.txt]: done

**Pull request merged**